apps/api/src/app/workflows-v2/usecases/validate-content/collect-placeholders/collect-placeholder-with-defaults.usecase.ts

```diff
@@ -11,16 +12,57 @@ export class CollectPlaceholderWithDefaultsUsecase {
     if (!command.controlValues) {
       return {};
     }
-    const placeholders: Record<string, PlaceholderAggregation> = {};
+    let placeholders: Record<string, PlaceholderAggregation> = {};
     const flattenedControlValues = flattenJson(command.controlValues);
 
-    for (const controlValueKey of Object.keys(flattenedControlValues)) {
-      const flattenedControlValue = flattenedControlValues[controlValueKey];
+    for (const [controlValueKey, flattenedControlValue] of Object.entries(flattenedControlValues)) {
       placeholders[controlValueKey] = this.extractPlaceholdersLogic(flattenedControlValue);
     }
 
+    if (command.origin === WorkflowOriginEnum.EXTERNAL) {
+      placeholders = this.filterPayloadVariables(placeholders);
+      const payloadPlaceholders: Record<string, PlaceholderAggregation> = this.buildPayloadPlaceholders(command);
+      placeholders = { ...placeholders, ...payloadPlaceholders };
+    }
+
     return placeholders;
   }
+
+  private buildPayloadPlaceholders(command: CollectPlaceholderWithDefaultsCommand) {
+    const payloadPlaceholders: Record<string, PlaceholderAggregation> = {
+      payloadPlaceholder: {
+        nestedForPlaceholders: {},
+        regularPlaceholdersToDefaultValue: {},
+      },
+    };
+
+    for (const [key, value] of Object.entries(command.payloadSchema?.properties || {})) {
+      payloadPlaceholders.payloadPlaceholder.regularPlaceholdersToDefaultValue[`{{payload.${key}}}`] =
+        typeof value === 'boolean' ? `{{payload.${key}}}` : (value.default as string);
+    }
+
+    return payloadPlaceholders;
+  }
+
+  private filterPayloadVariables(
+    placeholders: Record<string, PlaceholderAggregation>
+  ): Record<string, PlaceholderAggregation> {
+    const resultPlaceholders = { ...placeholders };
+    for (const placeholderAggregation of Object.values(resultPlaceholders)) {
+      const regularPlaceholders = placeholderAggregation.regularPlaceholdersToDefaultValue;
+      const nestedPlaceholders = placeholderAggregation.nestedForPlaceholders;
+
+      placeholderAggregation.regularPlaceholdersToDefaultValue = Object.fromEntries(
+        Object.entries(regularPlaceholders).filter(([key]) => !key.includes('{{payload.'))
+      );
+      placeholderAggregation.nestedForPlaceholders = Object.fromEntries(
+        Object.entries(nestedPlaceholders).filter(([key]) => !key.includes('{{payload.'))
+      );
+    }
+
+    return resultPlaceholders;
+  }
+
```

Comment:
I would expect all the work done here to be done at an additional step in the validator looking at the payload schema exmple vs the preview payload defaults generated